### ISSUE TITLE
Modeline VC branch name not updating on magit-checkout

### ISSUE THREAD

**Issue by Cwejman** [hlissner/doom-emacs#826]
Hey, great work! Doom has been my successful gateway to emacs and I'm loving it. Today though, I found out that the modeline branch name doesn't update when I checkout another branch (`magit-checkout`). Is this expected behaviour? If not please advice me on what information is necessary to further understand the issue.

In order to get auto-refreshed modelines in all buffers of the active workspace, my private config.el contains the following:

```elisp
(defun +cwejman-vc-refresh-modelines (a)
  (dolist (buffer (doom-buffer-list))
    (set-buffer buffer)
    (vc-refresh-state)
    (+doom-modeline--update-vcs)))

(advice-add 'magit-checkout :after #'+cwejman-vc-refresh-modelines)
```

**Comment by hlissner** [comment]
Thank you for the kind words! While this _is_ expected behavior, it is certainly not desirable! Essentially, magit leaves it to the user to inform vc of version-control changes to open buffers. 

In any case, this should be fixed as of 641cef9. Let me know if that isn't the case!

**Comment by seagle0128** [comment]
@hlissner Some one remind this might bring serious performance issue. 
> Unfortunately calling vc-refresh-state slows down staging/unstaging and branch switching in magit once a lot of buffers are opened.

Please refer to 
https://github.com/seagle0128/doom-modeline/commit/10ec55adcc69a08cbc3df814280f5e47f5f043c3#r30578517

I made a workaround [commit](https://github.com/seagle0128/doom-modeline/commit/f7a573906d8f58043c0bd09c3bbf4162c710055e) to address this issue, but not sure if it's applicable for you.
